app/database/models/rank.py

```diff
@@ -0,0 +1,50 @@
+from mongoengine import Document, IntField, LongField, StringField
+
+from discord.guild import Guild
+from discord.utils import get as find
+
+class Rank(Document):
+    level = IntField(null=False, required=True, primary_key=True, unique_with=["gid"])
+    gid = LongField(null=False, required=True)
+    role_id = LongField(null=False, required=True)
+    message = StringField()
+
+    def __str__(self):
+        return f"Rank(level={self.level}, gid={self.gid}, role_id={self.role_id})"
+
+    @classmethod
+    def get_rank(cls, guild: Guild, level:int = None, role_id:int|str = None, create:bool = False) -> "Rank":
+        """Get a member from the database.
```

Comment:
Copy-pasting the docs? Go to sleep 😡

Reply:
Funny enough, this was pretty early on, I just missed that line lol